codes/15686_py3_sjskgks.py

```diff
@@ -0,0 +1,35 @@
+from itertools import combinations
+import sys
+input = sys.stdin.readline
+
+sizeOfcity, min_ch = map(int, input().split()) # ch: chicken house
+posOfch = []
+posOfhouse = []
+for row in range(sizeOfcity):
+    col=0
+    for kind in map(int,input().split()):
+        if kind==1:
+            posOfhouse.append([row, col])
+        elif kind==2:
+            posOfch.append([row,col])
+        col+=1
+
+all_street_dist = [[0 for _ in range(len(posOfch))] for _ in range(len(posOfhouse))] # 집수 x 치킨집수 array
+for row in range(len(posOfhouse)):
+    h_row, h_col =posOfhouse[row]
+    for col in range(len(posOfch)):
+        ch_row, ch_col = posOfch[col]
+        all_street_dist[row][col] = abs(h_row-ch_row)+abs(h_col-ch_col)
+
+allcase = list(combinations([i for i in range(len(posOfch))], min_ch))
+temp_min=float("inf")
+min_dist = float("inf")
+for case in allcase:

```

Comment:
대입한 후 딱 한곳에서만 쓰이기 때문에 대입할 필요가 없는 것 같습니다.
또 리스트로 바꿀 이유도 없어 보입니다.
그리고 지금은 인덱스로 콤비네이션을 구하는데 더 좌표 자체로 콤비네이션을 구하면 더 깔끔할 것 같습니다. (아래 예시 코드에는 반영되어있지 않습니다.)
```suggestion
temp_min=float("inf")
min_dist = float("inf")
for case in combinations([i for i in range(len(posOfch))], min_ch):
```